integration-libs/cdc/core/store/effects/cdc-user-addresses.effect.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { CdcJsService } from '@spartacus/cdc/root';
+import {
+  Address,
+  GlobalMessageService,
+  GlobalMessageType,
+  UserActions,
+  UserAddressConnector,
+  UserAddressService,
+  UserIdService,
+} from '@spartacus/core';
+import { Observable, of } from 'rxjs';
+import { catchError, map, mergeMap, switchMap, take } from 'rxjs/operators';
+
+@Injectable()
+export class CdcUserAddressesEffects {
+  addressFieldKeys = ['line1', 'line2', 'region.name', 'town', 'postalCode'];
+
+  cdcAddUserAddress$: Observable<UserActions.UserAddressesAction> =
+    createEffect(
+      () =>
+        this.actions$.pipe(
+          ofType(UserActions.ADD_USER_ADDRESS_SUCCESS),
+          mergeMap(() => {
+            return this.updateDefaultAdressInCDC().pipe(
+              map((data: any) => {
+                return new UserActions.LoadUserAddressesSuccess(data);
+              }),
+              catchError((error) => {
+                let errorMessage = error?.errorMessage || ' ';
+                this.messageService.add(
+                  errorMessage,
+                  GlobalMessageType.MSG_TYPE_ERROR
+                );
+                return of(new UserActions.AddUserAddressFail(error));
+              })
+            );
+          })
+        ),
+      { dispatch: false } //prevent dispatching duplicate events in the case of CDC success / failure
```

Comment:
I can see in the implementation of the effect, that we return ngrx actions. Why we set here `{ dispatch: false }` ? (Same question applies to other effects in this file)

Reply:
hi @Platonn,


This is needed to prevent further event dispatches. User addresses effect already dispatches `UserActions.ADD_USER_ADDRESS_SUCCESS ` event, which would lead to the routing back to the address book page and invokes the get all Addresses. We do not want to repeat this network call for our implementation.

Reply:
If we don't want to trigger any other side effects (nor log success/failure of updating address in CDC), why do we return any ngrx actions? You said we don't want to dispatch them. Perhaps I'm missing something or lacking some knowledge about ngrx. Can you help me understand? 

```ts
              map((data: any) => {
                return new UserActions.LoadUserAddressesSuccess(data); // <---------
              }),
              catchError((error) => {
                let errorMessage = error?.errorMessage || ' ';
                this.messageService.add(
                  errorMessage,
                  GlobalMessageType.MSG_TYPE_ERROR
                );
                return of(new UserActions.AddUserAddressFail(error)); // <---------
              })
```


PS. If you don't want to trigger any other side effects, but only want to provide success/failure actions for producing events log, e.g. for ngrx dev tools, maybe you want to create your own Cdc-flavoured actions and dispatch them:
e.g. `CdcSetUserAddressesSuccess` and `CdcSetUserAddressesFail`?

Reply:
Good question @Platonn,

The reason we are returning is because the `createEffect` expects an Observable as the return type.
For the functionality we do not need to return anything.

Another alternative would be to return `return rxjs.EMPTY;`

What do you suggest here @Platonn ?

Reply:
I believe it's `catchError` that complains that you have to return an `Observable<never>` from the inner function. I believe `createEffect` doesn't complain, because it receives one big stream, which is an Observable by itself. So let's change `catchError` to `tap` with `error` callback `{ error: () => .... }`

Here is an example the the _add effect_. You can take it as an inspiration for all 3 effects.
```ts
  cdcUpdateUserAddress$ = createEffect(
    () =>
      this.actions$.pipe(
        ofType(UserActions.UPDATE_USER_ADDRESS_SUCCESS),
        mergeMap(() => {
          return this.updateDefaultAdressInCDC().pipe(
            tap({
              error: (error) => {
                let errorMessage = error?.errorMessage || ' ';
                this.messageService.add(
                  errorMessage,
                  GlobalMessageType.MSG_TYPE_ERROR
                );
              },
            })
          );
        })
      ),
    { dispatch: false }
```

Moreover, I suggest removing those obsolete comments then: ` //prevent dispatching duplicate events in the case of CDC success / failure`

Reply:
if this works, we can the signature and remove `{ dispatch: false }`

Reply:
@Platonn 

`createEffect` does expect the return type to be an Observable 

Attaching the JSdoc - 
```
(alias) createEffect<{
    dispatch: false;
}, false, unknown, Observable<UserActions.LoadUserAddressesSuccess | UserActions.AddUserAddressFail>>(source: () => Observable<UserActions.LoadUserAddressesSuccess | UserActions.AddUserAddressFail>, config?: Partial<...> | undefined): Observable<...> & CreateEffectMetadata
```


In this case, do you suggest that we return `return rxjs.EMPTY;` ?

Reply:
That's correct. `createEffect` expects the takes a function that should return an Observable.
Maybe I'm misunderstanding something. In which line you think we have to put `return rxjs.EMPTY;` ?

PS.
```ts
createEffect(
  // this is an arrow function that returns an Observable<any>
  () => this.actions$.pipe( ... <whatever logic inside> ... )
)
```

Reply:
IMHO we should fix this. But not crucial for 6.0. We can fix it in another PR.

Reply:
thanks @Platonn, will be adopting these changes too.